cmd/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	appcontroller "github.com/argoproj/argo-cd/cmd/argocd-application-controller/commands"
+	dex "github.com/argoproj/argo-cd/cmd/argocd-dex/commands"
+	reposerver "github.com/argoproj/argo-cd/cmd/argocd-repo-server/commands"
+	apiserver "github.com/argoproj/argo-cd/cmd/argocd-server/commands"
+	util "github.com/argoproj/argo-cd/cmd/argocd-util/commands"
+	cli "github.com/argoproj/argo-cd/cmd/argocd/commands"
+)
+
+func main() {
+	var command *cobra.Command
+
+	binaryName := filepath.Base(os.Args[0])
+	if val := os.Getenv("ARGOCD_BINARY_NAME"); val != "" {
```

Comment:
Rather than using environment variables for differentiating binaries, why not use subcommands like `argocd repo-server --more-repo-flags` and `argocd server --a-server-flag`?

From the community meeting, one concern was that we did not want users to have to type something like `argocd client` in order to access the client app, however I think cobra does have support for having both the root command be executable as well as subcommands

https://github.com/spf13/cobra#example

The root level command could be equivalent to the argocd client today, and you could still do things like `argocd server --foobar` to run the server component

Reply:
@dcherman Yes, we can add subcommands along with preserving the argocd cli behaviour. We can address this request as part of separate issue. Can you please create an issue to track it?